src/subscription_manager/exceptions.py

```diff
@@ -37,7 +37,8 @@
     "Remote server error. Please check the connection details, "
     "or see /var/log/rhsm/rhsm.log for more information."
 )
-BAD_CA_CERT_MESSAGE = _("Bad CA certificate: %s")
+BAD_CA_CERT_MESSAGE_GENERIC = _("Bad CA certificate: {file}")
+BAD_CA_CERT_MESSAGE_REASON = _("Bad CA certificate: {file}: {reason}")
```

Comment:
maybe it's me, however I don't see how "bad ca certificate: foo.pem: unknown" is more informative than not printing "unknown" at all; in both cases, there is no available detail on the failure happened when trying to use that foo.pem file.

Reply:
The difference is following. When we show that the reason is unknown, then it is obvious that the reason is really unknown.

When you do not show "reason: unknown", then it could like that we are just lazy to show any reason.

Reply:
Actually the case for "no SSL error" was there only because of:
https://github.com/candlepin/subscription-manager/blob/ed1d636df3fce7c58b20b6f23cceedd44b8865a6/src/rhsm/connection.py#L674-L678
and in tests; since the above situation is dead code (the FIXME comment there is spot-on), I added an extra commit to drop this dead code and simplify this case